app/src/main/java/org/oppia/android/app/player/state/StateFragmentPresenter.kt

```diff
@@ -195,6 +207,7 @@ class StateFragmentPresenter @Inject constructor(
   }
 
   fun onSubmitButtonClicked() {
+    recyclerViewAssembler.resetRawUserAnswer()
```

Comment:
Sorry to clarify: I understand why the answer needs to be reset. My confusion is why it's being done this way. I'm also confused by the last part of your explanation--I think you may be confusing real-time and submit-time errors. The latter only disable the submit button after it's clicked and an error is detected, and I think that case will be broken with this approach.

Reply:
Sorry @vrajdesai78 I'm not understanding--we _do_ use submit-time errors today. If a user triggers a submit-time error and rotates the device, per this code won't they lose their answer upon rotation? That's not what we'd want to happen in this case.

Reply:
As mentioned earlier, we do have problem with submit_time error and the app got crashed. So, as currently we don't have any submit-time errors can we go with this approach. May be we can fix this in the future when we fully support submit-time errors.

Reply:
Followed up on the other thread, but keeping this open until the problem is fully resolved.

Reply:
With this approach still for MathInteraction whenever I compute submit time error and rotate the screen, it is unable to reset the answer. I have also tried to reset answer in the top level of subscribeToAnswerOutcome as suggested by @BenHenning but still it's not working for MathInteraction.  For reference you can watch video below. 


https://user-images.githubusercontent.com/43074241/201206203-68185382-f306-4bc2-a622-e4a76bf7d56b.mp4

Reply:
@vrajdesai78 the latest changes don't include the replace() that was used instead of add() in ExplorationActivityPresenter--what happened to those? It was my understanding that this fixes nearly all of the configuration change issues, including the one you're showing in that video.

Reply:
I have used .replace() instead of .add() and still I am facing the same issue.